src/transformStateWithClones.js

```diff
@@ -64,8 +64,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const states = [];
+  let newState = { ...state };
+
+  for (const { operation, properties } of transforms) {
+    switch (operation) {
+      case 'addProperties':
+        for (const propAddKey in properties) {
+          newState[propAddKey] = properties[propAddKey];
+        }
```

Comment:
Иожно использовать спред оператор.
```suggestion
          newState = {...newState, ...properties};
```